tests/domains_test.go

```diff
@@ -1,34 +1,128 @@
+			Expect(err).NotTo(HaveOccurred())
+			Eventually(sess).Should(Say("=== %s Domains", testApp.Name))
+			Eventually(sess).ShouldNot(Say("%s", testApp.Name))
+			Eventually(sess).Should(Exit(0))
+		})
+
+		It("cannot add or remove domains", func() {
+			sess, err := start("deis domains:add %s --app=%s", domain, testApp.Name)
+			Expect(err).NotTo(HaveOccurred())
+			Eventually(sess.Err).Should(Say("500 Internal Server Error")) // better error/explanation needed from cli
+			Eventually(sess).Should(Exit(1))
+
+			sess, err = start("deis domains:remove %s --app=%s", domain, testApp.Name)
+			Expect(err).NotTo(HaveOccurred())
+			Eventually(sess.Err).Should(Say("500 Internal Server Error")) // better error/explanation needed from cli
+			Eventually(sess).Should(Exit(1))
+		})
+	})
+
 	Context("with a deployed app", func() {
+		var curlCmd Cmd
+		var cmdRetryTimeout int
 
-		XIt("can add, list, and remove domains", func() {
-			// "deis domains:list --app=%s", app
-			// "deis domains:add %s --app=%s", domain, app
-			// "deis domains:list --app=%s", app
-			// curl app at both root and custom domain
-			// "deis domains:remove %s --app=%s", domain, app
-			// "deis domains:list --app=%s", app
-			// curl app at both root and custom domain, custom should fail
+		BeforeEach(func() {
+			cmdRetryTimeout = 10
+			domain = getRandDomain()
+			testApp = deployApp("example-go")
 		})
 
-		XIt("can add, list, and remove certs", func() {
-			// "deis domains:add %s --app=%s", domain, app
-			// "deis certs:list", app
-			// "deis certs:add %s %s", certPath, keyPath
-			// wait for 60 seconds until cert generation is done?
-			// curl the custom SSL endpoint
-			// "deis certs:remove %s", domain
-			// "deis certs:list", app
-			// curl the custom SSL endpoint, should fail
+		AfterEach(func() {
+			defer os.Chdir("..")
+			destroyApp(testApp)
+		})
+
+		It("can add, list, and remove domains", func() {
+			sess, err := start("deis domains:list --app=%s", testApp.Name)
+			Expect(err).NotTo(HaveOccurred())
+			Eventually(sess).Should(Say("=== %s Domains", testApp.Name))
+			Eventually(sess).Should(Say("%s", testApp.Name))
+			Eventually(sess).Should(Exit(0))
+
+			sess, err = start("deis domains:add %s --app=%s", domain, testApp.Name)
+			Expect(err).NotTo(HaveOccurred())
+			Eventually(sess).Should(Say("Adding %s to %s...", domain, testApp.Name))
+			Eventually(sess).Should(Say("done"))
+			Eventually(sess).Should(Exit(0))
+
+			sess, err = start("deis domains:list --app=%s", testApp.Name)
+			Expect(err).NotTo(HaveOccurred())
+			Eventually(sess).Should(Say("=== %s Domains", testApp.Name))
+			Eventually(sess).Should(Say("%s", domain))
+			Eventually(sess).Should(Exit(0))
+
+			// curl app at both root and custom domain, both should return 200
+			curlCmd = Cmd{CommandLineString: fmt.Sprintf(`curl -sL -w "%%{http_code}\\n" "%s" -o /dev/null`, testApp.URL)}
+			Eventually(cmdWithRetry(curlCmd, strconv.Itoa(200), cmdRetryTimeout)).Should(BeTrue())
```

Comment:
if we're expecting status codes, should we use `http.StatusOK` instead of 200 to indicate that? https://golang.org/pkg/net/http/#pkg-constants